### PR TITLE
Carson/node streams

### PR DESCRIPTION
**Pull request by carsonfarmer**
Brings main up to date with latest release

### CODE REVIEW

**Comment by andrewxhill** [comment]
handing review off to @asutula i'm getting too behind here

**Review by asutula — APPROVED** [review]
Couple little questions, but cool!

File: src/storage.ts [hunk 1]
```diff
@@ -32,19 +39,18 @@ function initStorage(account: Account, options: { brokerInfo: BrokerInfo }) {
...
-      data: File | FormData,
+      data: File | Readable,
       options: OpenOptions = {}
     ): Promise<Request> {
-      let formData: FormData;
-      if (isFormData(data)) {
-        formData = data;
+      let body: FormData | Readable;
+      if (isReadableStream(data)) {
+        body = data;
```
[the remark this answers is not in the document]
Comment (carsonfarmer) [reply src/storage.ts:47]: Yeh, the trick is to `import type { Readable } from "streams"`, so the actual javascript code is never imported, so the compiled output doesn't contain any node-specific javascript.

File: src/storage.ts [hunk 2]
```diff
@@ -53,8 +59,9 @@ function initStorage(account: Account, options: { brokerInfo: BrokerInfo }) {
       });
       const res = await fetch(`${url}/upload`, {
         method: "POST",
-        body: formData,
+        body: body as BodyInit,
```
[the remark this answers is not in the document]
Comment (carsonfarmer) [reply src/storage.ts:62]: it comes from "dom" types (which are available globally).

**Pull request merged**